src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/OsmPbfCounter.java

```diff
@@ -69,6 +71,10 @@
     private final Set<Long> wayIdentifiersToInclude = new HashSet<>();
     private final Set<Long> relationIdentifiersToInclude = new HashSet<>();
 
+    // Keep track of included nodes so that they can be used in calculating if a way intersects the
+    // given shard
+    private final Map<Long, Location> nodeIdentifierToLocation = new HashMap<>();
```

Comment:
What do you think about combining `nodeIdentifiersToInclude` and `nodeIdentifierToLocation` to be a single Map. That way we aren't tracking the identifiers and the identifiers + locations in two separate collections?

Reply:
Hmm my comment is misleading, it should read "Keep track of ALL nodes...". `nodeIdentifiersToInclude` tracks only the IDs of nodes that should be included, whereas `nodeIdentifierToLocation` includes all nodes from the pbf (basically a node location store). This is because we need to use the location of nodes beyond the shard boundary to check for intersections. The two sets have different node spaces, so I think they shouldn't be combined to a single map.

Reply:
This makes sense, thanks. It's essentially every location in the PBF.